vault/identity_store_aliases.go

```diff
@@ -256,6 +247,51 @@ func (i *IdentityStore) handleAliasUpdateCommon(req *logical.Request, d *framewo
 		return nil, err
 	}
 
+	var existingEntity *identity.Entity
+	var lockKeys []string
+	if !newAlias {
+		// Verify that the combination of alias name and mount is not
+		// already tied to a different alias
+		if aliasByFactors != nil && aliasByFactors.ID != alias.ID {
+			return logical.ErrorResponse("combination of mount and alias name is already in use"), nil
+		}
+
+		// Fetch the entity to which the alias is tied to
+		existingEntity, err = i.MemDBEntityByAliasID(alias.ID, true)
+		if err != nil {
+			return nil, err
+		}
+
+		if existingEntity == nil {
+			return nil, fmt.Errorf("alias is not associated with an entity")
+		}
+		lockKeys = append(lockKeys, existingEntity.ID)
+	}
+
+	if canonicalID != "" {
```

Comment:
If you move lockKeys up to the top and add in the canonicalID up then, if we have one, you can pull the LocksForKeys and lock/unlock lines out of this if/else